build_scripts/api/dreambooth/ui_functions.py

```
@@ -0,0 +1,1136 @@
+        else:
+            return 2, -res[0]
+
+    sorted_counts = sorted(
+        out_counts.items(), key=lambda x: sort_key(x[0]), reverse=True
+    )
+    total_images = 0
+    for res, count in sorted_counts:
+        total_images += count
+        print(
+            f"RES: {res}  -  {count}  - {[os.path.basename(file) for file in out_paths[res]]}"
+        )
+
+    out_images = []
+    out_status = ""
+
+    if not dry_run:
+        if not os.path.exists(dest_dir):
+            os.makedirs(dest_dir)
+    pbar.set_description("Cropping images.")
+
+    pbar.reset(total_images)
+    for res, images in out_paths.items():
+        for image in images:
+            pbar.update()
+            out_img = os.path.join(
+                dest_dir, os.path.splitext(os.path.basename(image))[0] + ".png"
+            )
+            cropped = open_and_trim(image, res, True)
+            if not dry_run:
+                print(f"\nSaving to {out_img}")
+                cropped.save(out_img, quality=100)
+            out_images.append(cropped)
+            shared.status.current_image = [cropped]
+
+        out_status = (
+            f"{'Saved' if not dry_run else 'Previewed'} {total_images} cropped images."
+        )
+    status.end()
+    return out_status, out_images
+
+
+def create_model(
+        new_model_name: str,
+        ckpt_path: str,
+        shared_src: str,
+        from_hub=False,
+        new_model_url="",
+        new_model_token="",
+        extract_ema=False,
+        train_unfrozen=False,
+        model_type="v1"
+):
+    printm("Extracting model.")
+    res = 512
+    is_512 = model_type == "v1"
+    if model_type == "v1x" or model_type=="v2x-512":
+        res = 512
+    elif model_type == "v2x":
+        res = 768
+    elif model_type == "sdxl":
+        res = 1024
+    sh = None
+    try:
+        from core.handlers.status import StatusHandler
+        sh = StatusHandler()
+        sh.start(5, "Extracting model")
+    except:
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are generically passing an `Exception` object without performing any other operation on it. This may hide error conditions that can otherwise be quickly detected and addressed. We recommend that you catch a more specific exception. If the code must broadly catch all exceptions, consider logging the stack trace using the [logging.exception()](https://docs.python.org/3/library/logging.html#logging.exception) API. For example,

```
try:
    x = 1 / 0
except ZeroDivisionError as e:
    logging.exception(&#39;ZeroDivisionError: %s&#39;, e)

```

Similar issue at line number 997.